packages/core/datadog-sourcemaps.gradle

```diff
@@ -203,10 +205,19 @@ private def runShellCommand(
 }
 
 private def getServiceName(variant) {
-    if (project.hasProperty("datadog")) {
+    if (project.hasProperty("datadog") && project.datadog.hasProperty("serviceName")) {
         if (project.datadog.serviceName) {
```

Comment:
isn't this check redundant then? it is Groovy, so here it is implicit. If it checks only for `null`, then it is the same as `project.datadog.hasProperty("serviceName")` the line above. If it does also check for `empty` in case of `String`, then it is not the same.

Reply:
It also checks for `empty`.

See here: https://onecompiler.com/groovy/42e5tdbyz

```groovy
def properties = [
  'serviceName': ''
]

def serviceName = 'default'

if (properties.serviceName) {
  serviceName = properties.serviceName
}

println "Service Name: " + serviceName
```
Output is:

```sh
Service Name: default
```

I should double check the logic with a quick test. The reason why I have introduced this change is because I was getting an empty/null `serviceName` when specifying other properties in `datadog` (while it works if you don't use the `datadog` property at all).

Reply:
ok, makes sense indeed. This is why I don't like Groovy - too many implicit things.